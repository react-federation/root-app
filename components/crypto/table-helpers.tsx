import React, { VoidFunctionComponent } from 'react'
import { Link, useRouteMatch } from 'react-router-dom'
import { Text } from '@chakra-ui/layout'

export interface TableHelperProps {
  value: string | number;
  row?: any;
}

export const PriceFormatter: VoidFunctionComponent<TableHelperProps> = ({ value }) => {
  const parsedValue = parseFloat(value as string).toLocaleString('en-US')

  return <Text>USD {parsedValue}</Text>
}

export const RenderLink: VoidFunctionComponent<TableHelperProps> = ({ row, value }) => {
  const { url } = useRouteMatch()

  const to = `${url}/${row.id}`

  return <Link to={to}>{value}</Link>
}

export const RenderIcon: VoidFunctionComponent<TableHelperProps> = ({ value }) => (
  <img src={value as string} width="24px" />
)
