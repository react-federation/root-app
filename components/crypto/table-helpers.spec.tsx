import { BrowserRouter } from 'react-router-dom'
import setup from 'utils/test/setup'
import { PriceFormatter, RenderLink, RenderIcon } from './table-helpers'

describe('PriceFormatter', () => {
  it('should match snapshot', () => {
    const { asFragment } = setup(PriceFormatter, { value: 1000 })

    expect(asFragment()).toMatchSnapshot()
  })
})

describe('RenderLink', () => {
  it('should match snapshot', () => {
    const { asFragment } = setup(RenderLink, { value: 1000, row: { id: 'btc' } }, BrowserRouter)

    expect(asFragment()).toMatchSnapshot()
  })
})

describe('RenderIcon', () => {
  it('should match snapshot', () => {
    const { asFragment } = setup(RenderIcon, { value: 'http://image.test/logo.jpeg' })

    expect(asFragment()).toMatchSnapshot()
  })
})
