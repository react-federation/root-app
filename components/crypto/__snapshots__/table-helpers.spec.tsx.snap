// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceFormatter should match snapshot 1`] = `
<DocumentFragment>
  <p
    class="chakra-text emotion-0"
  >
    USD 1,000
  </p>
</DocumentFragment>
`;

exports[`RenderIcon should match snapshot 1`] = `
<DocumentFragment>
  <img
    src="http://image.test/logo.jpeg"
    width="24px"
  />
</DocumentFragment>
`;

exports[`RenderLink should match snapshot 1`] = `
<DocumentFragment>
  <a
    href="//btc"
  >
    1000
  </a>
</DocumentFragment>
`;
